### PR TITLE
remove bokeh.charts from the Bokeh core library

### LINKED ISSUE #5246: py.test prints garbage when there is a syntax error in *.py files
By garbage I mean a dense block of text instead of a nicely formatted traceback. If you are patient enough, you can figure out what's wrong with your code, but it would be nice to get the usual output.

### PR DESCRIPTION
**Pull request by bryevdv**
@canavandl @mattpap please look over this as soon as possible. 

@bokeh/dev The removes `bokeh.charts` and all references from the Bokeh core project. In the near future, the https://github.com/bokeh/bkcharts repo will be completed and a new `bkcharts` package made available Once this occurs, there are still some actions to perform before the next Bokeh release:

* package and release `bkcharts` to `defaults` and pypi
* make Bokeh depend on `bkcharts` 
* uncomment temporary `bkcharts` imports in the core library
* deploy and link to `bkcharts` docs from bokeh docs

The decision to go ahead with the removal before `bkcharts` is compelte was taken because a new version of Pandas is breaking `bokeh.charts` tests. 

fixes #5246

### CODE REVIEW

File: bokeh/charts/__init__.py [hunk 1]
```diff
@@ -1,40 +1,4 @@
...
-from .builders.chord_builder import Chord
-
-# easy access to required bokeh components
-from ..models import ColumnDataSource; ColumnDataSource
-from ..io import curdoc, output_file, output_notebook, reset_output, save, show, gridplot
-
-# Silence pyflakes
-(curdoc, output_file, output_notebook, reset_output, save, show, gridplot)
+# TODO (bev) restore shim when bkcharts package is ready
+# from bkcharts import * # NOQA
```
Comment (mattpap) [inline bokeh/charts/__init__.py]: Should we add deprecation message at this point, pointing people to preferably use `bkcharts`?

Comment (bryevdv) [inline bokeh/charts/__init__.py]: Yes I will add that when the `bkcharts` package is available and this import is uncommented

**Comment by mattpap** [comment]
There are a few files that still contain references to charts that possibly should be removed:

1. `scripts/interactive_tester`
2. `examples/howto/layouts/words_and_plots.py`

Maybe more, though I'm fine with merging this PR and cleaning up later.

**Comment by mattpap** [comment]
```
tests/examples/test_examples.py::test_file_examples[/home/travis/build/bokeh/bokeh/examples/howto/layouts/words_and_plots.py-example91] 
[INFO] Example run in 0.528s
[PY] Traceback (most recent call last):
[PY]   File "<string>", line 13, in <module>
[PY]   File "/home/travis/build/bokeh/bokeh/examples/howto/layouts/words_and_plots.py", line 4, in <module>
[PY]     from bokeh.charts import Scatter
[PY] ImportError: cannot import name Scatter
```

**Comment by bryevdv** [comment]
@mattpap I have removed those as well as some few others.

File: examples/howto/prevent_user_zooming_or_panning_too_far.py [hunk 2]
```diff
@@ -75,24 +74,6 @@
...
-plot_cat_bounded = HeatMap(
-    fruits, y='year', x='fruit', values='fruit_count', stat=None,
-    title="Heatmap with bounds x:['apples', 'pears'], y:['2009', '2010', '2013']"
-)
-###### -- ranges set here -- ########
-plot_cat_bounded.x_range.bounds = ['apples', 'pears']
-plot_cat_bounded.y_range.bounds = ['2009', '2010', '2013']
-###### -- end -- ########
-
-
```
Comment (bryevdv) [inline examples/howto/prevent_user_zooming_or_panning_too_far.py]: These should be replaced with a handmade heat map at some point.

**Comment by bryevdv** [comment]
The other good news is this raises our unit test coverage to 88%

**Comment by bryevdv** [comment]
@canavandl @mattpap will merge at 15:00 CST unless I hear otherwise.

**Comment by mattpap** [comment]
> The other good news is this raises our unit test coverage to 88%

And resolves #5246.

**Comment by canavandl** [comment]
I grepped around and couldn't find any remaining references and the tests pass for me locally.

**Pull request merged**